### PR TITLE
Add LLVM/Clang 12.0.0 toolchain

### PR DESCRIPTION
**Pull request by supergrecko**
Adds LLVM 12.0.0's toolchain to the Compiler Explorer compiler list. First time submitting a patch, let me know if there's anything missing!

The patch is currently untested as the release is not out yet.

I imagine we could also consider changing the --gcc-toolchain to GCC 10.3

Infra: https://github.com/compiler-explorer/infra/pull/496

### CODE REVIEW

**Comment by ojeda** [comment]
Please add the libs too, see e.g. https://github.com/compiler-explorer/compiler-explorer/commit/3e1268143f97e056436d45b9b875fafda70bd714.

Thanks!

File: etc/config/assembly.amazon.properties [hunk 1]
```diff
@@ -190,6 +190,7 @@ compiler.llvmas1100.exe=/opt/compiler-explorer/clang-11.0.0/bin/llvm-mc
 compiler.llvmas1100.semver=11.0.0
 compiler.llvmas1101.exe=/opt/compiler-explorer/clang-11.0.1/bin/llvm-mc
 compiler.llvmas1101.semver=11.0.1
+compiler.llvmas1200.exe=/opt/compiler-explorer/clang-12.0.0/bin/llvm-mc
```
Comment (ojeda) [inline etc/config/assembly.amazon.properties:193]: Missing the `semver` key.

**Comment by supergrecko** [comment]
> Please add the libs too, see e.g. [3e12681](https://github.com/compiler-explorer/compiler-explorer/commit/3e1268143f97e056436d45b9b875fafda70bd714).
> 
> Thanks!

Thank you, the llvm library keys have been updated now.

File: etc/config/assembly.amazon.properties (hunk 1 above)
Comment (supergrecko) [reply etc/config/assembly.amazon.properties:193]: Must've missed that, thank you

**Comment by mattgodbolt** [comment]
Looks like clang 12 is out! I'm building it now, and will then review and merge your PRs!

**Review by mattgodbolt — APPROVED** [review]
Looks great; thanks!

**Pull request merged**